### ISSUE TITLE
Broken test data in valid-service-1.x.json

### ISSUE THREAD

**Issue by tokcum** [CycloneDX/specification#294]
Hi,

working on [cyclonedx-rust-cargo to support 1.4](https://github.com/CycloneDX/cyclonedx-rust-cargo/pull/399) I'm experiencing failed integration tests when verifying against valid-service-1.4.json while valid-service-1.4.xml just works.

The error tells me that a dependency references a bom which is not available in the file. Comparing the JSON to the XML file, I found that in JSON, the bom-ref is different. Setting the bom-ref to the value referenced in ref (like it is in XML), the test works fine.

```
"components": [
    {
      "bom-ref": "pkg:npm/acme/component@1.0.0",
      [...]

"dependencies": [
    {
      "ref": "pkg:maven/com.acme/stock-java-client@1.0.12",
```

In XML it looks like this:

```
<components>
        <component type="library" bom-ref="pkg:maven/com.acme/stock-java-client@1.0.12">
        [...]

<dependencies>
        <dependency ref="pkg:maven/com.acme/stock-java-client@1.0.12">
```

It looks like this derivation existed from the very beginning of the test data. So, maybe I'm missing something or the implementation in cyclonedx-rust-cargo is stricter than other projects using the spec's test data.

I've prepared a fix in my own fork. If my analysis above is correct, let me know. I will create a pull request. Thanks.

**Comment by jkowalleck** [comment]
Thank you very much for the report, @tokcum .

> I've prepared a fix in my own fork. If my analysis above is correct, let me know. I will create a pull request. Thanks.

Could you link the find version you used?
Or even provide a pullrequest wit the fix that worked for you?

**Comment by tokcum** [comment]
@jkowalleck, great. I've created a PR.

**Comment by jkowalleck** [comment]
LGTM

**Comment by jkowalleck** [comment]
:heavy_check_mark: issue was validated
:checkered_flag: issue was fixed via #295
:checkered_flag: fix was forward-ported to v1.6 via #297
